File: Top 100 Codes/02-Working with Numbers/15-Addition of two fractions/Additionoftwofractions.cpp
#include <iostream>
using namespace std;

// Function to find the Greatest Common Divisor (GCD) using the iterative approach
int findGCD(int a, int b) {
    int gcd;
    for (int i = 1; i <= a && i <= b; i++) {
        if (a % i == 0 && b % i == 0)
            gcd = i;
    }
    return gcd;
}

int main() {
    int num1, den1, num2, den2;

    // Taking user input for the first fraction
    cout << "Enter numerator and denominator of the first fraction: ";
    cin >> num1 >> den1;

    // Taking user input for the second fraction
    cout << "Enter numerator and denominator of the second fraction: ";
    cin >> num2 >> den2;

    // Finding the Least Common Multiple (LCM) of the denominators
    int lcm = (den1 * den2) / findGCD(den1, den2);

    // Computing the sum of fractions by making denominators equal
    int numeratorSum = (num1 * (lcm / den1)) + (num2 * (lcm / den2));

    // Simplifying the result
    int gcdResult = findGCD(numeratorSum, lcm);
    int simplifiedNumerator = numeratorSum / gcdResult;
    int simplifiedDenominator = lcm / gcdResult;

    // Displaying the result
    cout << num1 << "/" << den1 << " + " << num2 << "/" << den2
         << " = " << simplifiedNumerator << "/" << simplifiedDenominator << endl;

    return 0;
}

File: Top 100 Codes/02-Working with Numbers/15-Addition of two fractions/Additionoftwofractions.py
def findGCD(a, b):
    """
    Function to find the Greatest Common Divisor (GCD) using an iterative approach.
    """
    gcd = 1
    for i in range(1, min(a, b) + 1):
        if a % i == 0 and b % i == 0:
            gcd = i
    return gcd


# Taking user input for the first fraction
num1, den1 = map(int, input("Enter numerator and denominator of first fraction (separated by space): ").split())

# Taking user input for the second fraction
num2, den2 = map(int, input("Enter numerator and denominator of second fraction (separated by space): ").split())

# Finding the Least Common Multiple (LCM) of the denominators
lcm = (den1 * den2) // findGCD(den1, den2)

# Computing the sum of fractions by making denominators equal
numerator_sum = (num1 * (lcm // den1)) + (num2 * (lcm // den2))

# Simplifying the resulting fraction
gcd_result = findGCD(numerator_sum, lcm)
simplified_numerator = numerator_sum // gcd_result
simplified_denominator = lcm // gcd_result

# Displaying the result
print(f"{num1}/{den1} + {num2}/{den2} = {simplified_numerator}/{simplified_denominator}")
